File: Baekjoon/딕셔너리/11652.py
# https://www.acmicpc.net/problem/11652

import sys
input = sys.stdin.readline

n = int(input())
dic = {}

for i in range(n) :
    card = int(input())
    if card in dic :
        dic[card] += 1
    else :
        dic[card] = 1

answer = sorted(dic.items(), key = lambda x : (-x[1], x[0]))
print(answer[0][0])

File: Baekjoon/딕셔너리/11652.cpp
// https://www.acmicpc.net/problem/11652

#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    int n;
    cin >> n;
    vector<long long> v(n);

    for (int i = 0; i < n; i++) {
        cin >> v[i];
    }

    sort(v.begin(), v.end());

    long long answer = v[0];
    int cnt = 1;
    int max_cnt = 1;

    for (int i = 1; i < n; i++) {
        if (v[i] == v[i - 1]) {
            cnt++;
        } else {
            cnt = 1;
        }

        if (cnt > max_cnt) {
            max_cnt = cnt;
            answer = v[i];
        }
    }

    cout << answer << '\n';

    return 0;
}